lib/_http_client.js

```diff
@@ -138,7 +138,11 @@ function ClientRequest(options, cb) {
       timeout: self.timeout,
       rejectUnauthorized: !!options.rejectUnauthorized
     };
-    const newSocket = self.agent.createConnection(optionsPath, oncreate);
+    const newSocket = self.agent
+      ? self.agent.createConnection(optionsPath, oncreate)
+      : typeof options.createConnection === 'function'
+        ? options.createConnection(optionsPath, oncreate)
+        : net.createConnection(optionsPath, oncreate);
```

Comment:
We don't need to pass `oncreate` to `net.createConnection()`, it always immediately returns a socket.

Reply:
Thanks, fixed.